### PR TITLE
Implement `disabled`, `isPreferred` and `documentation` fields for code actions

### LINKED ISSUE #9989: Implement missing `CodeAction` fields
### Feature Description:
* Add the `isPreferred` and `disabled` fields in `CodeAction` 
* Add the `documentation` field in `CodeActionMetadataProvider `

(https://code.visualstudio.com/api/references/vscode-api#CodeAction). AndWith a bit of luck, the semantics are already implemented through the editor. In particular menu enablement might not be.

### PR DESCRIPTION
**Pull request by lucas-koehler**
Fixes #9989
Closes #10073 - this builds on #10073 but also hands in the documentation metadata into monaco.

Contributed on behalf of STMicroelectronics

Signed-off-by: Lucas Koehler <lkoehler@eclipsesource.com>
#### What it does
Fixes https://github.com/eclipse-theia/theia/issues/9989

* Adds support for the `disabled` and `isPreferred` fields to the `CodeAction` interface
* Adds support for the `documentation` field to `CodeActionProviderMetaData` interface.

#### How to test
* Include my [code actions test plugin](https://github.com/lucas-koehler/vscode-extension-samples/releases/download/code-actions-sample-0.0.3/code-actions-sample-0.0.3.vsix). For reference, it's code can be found [here](https://github.com/lucas-koehler/vscode-extension-samples/tree/code-actions-sample-0.0.3/code-actions-sample)
* Start the application and open a markdown file
* Type `:)`
* Lightbulb should be blue to indicate a preferred fix
* Open quick fix menu and look at the entries
  * The poop emoji entry is hidden (present on master).
  * There should be an entry `Learn more about Emojis...` which opens the emoji wikipedia page when selected. This is the documentation entry

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/plugin/languages.ts [hunk 1]
```diff
@@ -446,13 +449,26 @@ export class LanguagesExtImpl implements LanguagesExt {
         metadata?: theia.CodeActionProviderMetadata
     ): theia.Disposable {
         const callId = this.addNewAdapter(new CodeActionAdapter(provider, this.documents, this.diagnostics, pluginModel ? pluginModel.id : '', this.commands));
+
+        let documentation: CodeActionProviderDocumentation | undefined;
+        let disposables: DisposableCollection | undefined;
+        if (metadata && metadata.documentation) {
+            disposables = new DisposableCollection();
+            documentation = metadata.documentation.map(doc => ({
+                kind: doc.kind.value!,
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/plugin/languages.ts:458]: @lucas-koehler I believe that [`value`](https://code.visualstudio.com/api/references/vscode-api#CodeActionKind) is non-optional in vscode, should we align?

- [vscode@master](https://github.com/microsoft/vscode/blob/25ee7d964c45b42d42c7afdc8bf619550cb72a17/src/vscode-dts/vscode.d.ts#L2216-L2219)
- [vscode@1.53.2](https://github.com/microsoft/vscode/blob/622cb03f7e070a9670c94bae1a45d78d7181fbd4/src/vs/vscode.d.ts#L2127-L2130)

**Review by vince-fugnitto — COMMENTED** [review]
I confirmed that functionally the changes work very well :+1: 
- [x] confirmed that disabled items are in fact disabled compared to master
- [x] confirmed that documentation entries work well

**Comment by lucas-koehler** [comment]
@vince-fugnitto Thanks for your review :) I aligned CodeActionKind's `value` property to be non-optional and removed the now obsolete `!` operators.

**Review by msujew — APPROVED** [review]
The changes are looking good to me!

- [x] Preferred code actions are highlighted
- [x] Clicking the documentation code actions brings up the pages in the browser

I noticed that a separator is missing in Theia compared to VSCode:

image

Theia:

image

I assume that's unrelated to your changes. Or do you think you can "simply" add that as well?

**Comment by lucas-koehler** [comment]
> I noticed that a separator is missing in Theia compared to VSCode:
> [...]
> I assume that's unrelated to your changes. Or do you think you can "simply" add that as well?
@msujew Unfortunately, as this menu comes from monaco, I do not know how to add this in a simple way.

**Comment by msujew** [comment]
@lucas-koehler Completely fine with me. Since Colin is working on updating Monaco anyway, it's not that important 👍

**Comment by JonasHelming** [comment]
@vince-fugnitto : Fine with merging this?

**Pull request merged**